autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -207,9 +207,11 @@ async def execute_node(
 
         # Update execution stats
         if execution_stats is not None:
-            execution_stats = execution_stats.model_copy(
-                update=node_block.execution_stats.model_dump()
-            )
+            # The iteration set instead of using the model_copy is intentional,
+            # as we want to update the stats in place.
+            for key, value in node_block.execution_stats.model_dump().items():
+                if hasattr(execution_stats, key):
+                    setattr(execution_stats, key, value)
```

Comment:
This looks a bit janky, for this kind of use I'd advise implementing `+=` operator on `NodeExecutionStats`. Looks more intuitive :)

Reply:
Yes it will overwrite all attributes, which the previous implementation is doing too.

> If you want to get execution_stats out of the function, why not return it?
Then we will do the exact same thing from all the callers, which basically merging/overwriting the stats from the existing stats

Reply:
Also it's hard to make a neat code that returns an object + async generator at the same time. Hence passing it as a param to be mutated seems to be cleanest approach.